### PR TITLE
first pass linting changes

### PR DESCRIPTION
**Pull request by duranb**
I've added some linting and made changes to fix any errors that came up. @AaronPlave feel free to update the rules if you find that it's too aggressive/not aggressive enough

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    "no-unused-vars": "off",
+    "@typescript-eslint/no-unused-vars": "error",
+    "no-use-before-define": "off",
+    // 'nofunc' option means function hoisting is still allowed
+    "@typescript-eslint/no-use-before-define": ["error", "nofunc"],
+    "no-shadow": "off",
+    "@typescript-eslint/no-shadow": ["error"],
+
+    // >>> style rules carried over from tslint
+    // todo: finish going through tslint-old.json and transferring new rules over
```
Comment (AaronPlave) [inline .eslintrc.json]: Is this still a TODO?

Comment (duranb) [reply .eslintrc.json]: oops...i forgot to clean this up.

**Review by AaronPlave — APPROVED** [review]

File: src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -38,7 +80,7 @@ export const Modal = (props: ModalProps) => {
     title,
     // description,
     children,
-    actions = [],
+    // actions = [],
```
Comment (duranb) [inline src/components/Modal/Modal.tsx]: @AaronPlave was this an okay change? I wasn't sure where `actions` was supposed to be used.

Comment (AaronPlave) [reply src/components/Modal/Modal.tsx]: Oh it looks like actions and description aren't used at all, can get rid of those.

**Pull request merged**